src/zen/split-view/ZenViewSplitter.mjs

```diff
@@ -63,6 +63,304 @@ class SplitNode extends SplitLeafNode {
   }
 }
 
+class ZenSplitViewLinkDrop {
+  #zenViewSplitter;
+  _linkDropZone = null;
+  _isLinkDragging = false;
+
+  #handleLinkDragEnter;
+  #handleLinkDragLeave;
+  #handleLinkDragDrop;
+  #handleLinkDragEnd;
+
+  constructor(zenViewSplitter) {
+    this.#zenViewSplitter = zenViewSplitter;
+  }
+
+  init() {
+    this.#handleLinkDragEnter = this._handleLinkDragEnter.bind(this);
+    this.#handleLinkDragLeave = this._handleLinkDragLeave.bind(this);
+    this.#handleLinkDragDrop = this._handleLinkDragDrop.bind(this);
+    this.#handleLinkDragEnd = this._handleLinkDragEnd.bind(this);
+
+    const tabBox = document.getElementById('tabbrowser-tabbox');
+
+    tabBox.addEventListener('dragenter', this.#handleLinkDragEnter, true);
+    tabBox.addEventListener('dragleave', this.#handleLinkDragLeave, false);
+    tabBox.addEventListener('drop', this.#handleLinkDragDrop, false);
+    tabBox.addEventListener('dragend', this.#handleLinkDragEnd, false);
+  }
+
+  _createLinkDropZone() {
+    if (this._linkDropZone) return;
+
+    this._linkDropZone = document.createXULElement('box');
+    this._linkDropZone.id = 'zen-drop-link-zone';
+
+    const content = document.createXULElement('vbox');
+    content.setAttribute('align', 'center');
+    content.setAttribute('pack', 'center');
+    content.setAttribute('flex', '1');
+
+    const text = document.createXULElement('description');
+    text.setAttribute('value', 'Drop link to split'); // Localization! data-l10n-id
+
+    content.appendChild(text);
+    this._linkDropZone.appendChild(content);
+
+    this._linkDropZone.addEventListener('dragover', (event) => {
+      event.preventDefault();
+      event.stopPropagation();
+      event.dataTransfer.dropEffect = 'link';
+      if (!this._linkDropZone.hasAttribute('has-focus')) {
+        this._linkDropZone.setAttribute('has-focus', 'true');
+      }
+    });
+
+    this._linkDropZone.addEventListener('dragleave', (event) => {
+      event.stopPropagation();
+      if (!this._linkDropZone.contains(event.relatedTarget)) {
+        this._linkDropZone.removeAttribute('has-focus');
+      }
+    });
+
+    this._linkDropZone.addEventListener('drop', this._handleDropForSplit.bind(this));
+
+    const tabBox = document.getElementById('tabbrowser-tabbox');
+    tabBox.appendChild(this._linkDropZone);
+  }
+
+  _showLinkDropZone() {
+    if (!this._linkDropZone) this._createLinkDropZone();
+
+    this._linkDropZone.setAttribute('enabled', 'true');
+  }
+
+  _hideLinkDropZone(force = false) {
```

Comment:
I feel like we should completely remove the drop zone here, instead of just hiding. Firefox doesn't like things being rendered on top of the webview, even if it's hidden. So, only show the zone ONCE we are dragging, and removing it once done